Pull improvements from KVis fork

annehaley opened the pull request:
In summary, this PR includes the following changes which were merged into the master branch of the KVis fork after my own review. Below is a list of the changes.

1. Disable unreliable `codecov` and `pyppeteer` ci tests
2. Add new `playwright` tests to replace `pyppeteer` tests
3. Remove `direct-vuex` from the store
4. Migrate from `vue 2.6` to `2.7`, which has the composition api built in
5. Refactor any component files not already using composition api
6. General style and format cleanup

davidshq committed:
Create .devcontainer/devcontainer.json

davidshq committed:
Merge branch 'master' into reimplement-changes

davidshq committed:
Update gitconfig/exclusions; reorder dependencies

davidshq committed:
Upgrading Webpack from 4.x to 5.x and some Vue CLI packages

- Vue CLI packages upgraded because older versions of the Vue CLI don't support 5.x
- Change 'port' to 'host' in Vue config (per Webpack 5.x requirements)
- Remove port from 'PUBLIC_ADDRESS' in .env.development so it will work with 'host'.

davidshq committed:
Upgrading core-js to supported version and update ESLint and most dependents of ESLint

- Add Vuetify ESLint plugin. Note that this needs to stay at 1.x, the 2.x release is for Vuetify 3.

davidshq committed:
Removing deprecated `babel-eslint` and replacing with `@babel/eslint-parser`

davidshq committed:
Upgrade Vuetify from 2.6.3 to 2.6.14

davidshq committed:
Merge branch 'master' into reimplement-changes

davidshq committed:
Upgrading Vuex, updating ESLint config

- Ensure that both `@vue/airbnb` and `@vue/typescript/recommended` are enabled
- Turn off ESLint rule regarding providing file extensions when importing

davidshq committed:
Adding Jest for unit testing

davidshq committed:
Adding TS import resolver to do away with errors when importing

For example '@/django' previously threw an error

davidshq committed:
Minor linting changes

davidshq committed:
Creating `storeConfig` that is consumed by `createDirectStore` and can be used for unit testing

Minor formatting updates, removing `rootActionContext`, `moduleActionContext`, `rootGetterContext`, `moduleGetterContext` as these appear to be unused

Note: We don't want to mount the actual MIQA store when performing tests, rather we want to create an instance specifically for testing. This isn't easy to accomplish when the store is exported as already implemented. By separating the declaration and implementation of the store (using `storeConfig`) `createDirectStore` can still export `store` as the MIQA store instance while Jest can also pull in the `storeConfig` to implement its own testing instance.

davidshq committed:
Updating Jest configs

Note: Some of our node modules are not in the expected form. Usually Jest ignores the contents of the `node_modules` folder but in our case we need it to interact with a few directly, they are listed in the config file under `transformIgnorePatterns`

Note: Some file formats are not natively supported by Jest (e.g. most static assets), using `moduleNameMapper` we can have `jest-transform-stub` stub them out so they don't cause issues

davidshq committed:
Adding basic unit test of `setCurrentVtkIndexSlices`

davidshq committed:
Minor code refactoring, mainly using primitive types instead of object types

Note: See https://betterprogramming.pub/how-to-use-lowercase-types-for-primitives-and-objects-in-typescript-ddd00273849a

davidshq committed:
Removing unneeded pug related dependencies

davidshq committed:
Merge pull request #9 from davidshq/reimplement-changes

Baseline Improvements to MIQA Annotate, mainly JS/TS Tooling Oriented

davidshq committed:
Correcting tsconfig to target Vue 2 rather than 3, add `@vue/runtime-dom` which provides typing for Vue

davidshq committed:
Removing `direct-vuex`, add additional type defs

davidshq committed:
Minor code reformatting, specifying `lang="ts"` where appropriate

davidshq committed:
Refactoring to standard Vuex style instead of `direct-vuex`

davidshq committed:
Refactoring to standard Vuex style instead of `direct-vuex`, type `storeConfig` explicitly

davidshq committed:
Update types

davidshq committed:
Add missing properties/types to types, correct scans from having value `string[]` to `Scan`.

davidshq committed:
Reorging types roughly by alphabetical order (User/Email at beginning, MIQAStore at end) to allow easier navigation between types

davidshq committed:
Explicitly define permissions object

davidshq committed:
Remove `MIQAStore` as type from `initState`, TODO regarding windowLocked

davidshq committed:
Explicitly declaring `any` type on VRangeSlider

`options` is part of an undocumented API and thus no TS types have been defined, see: https://github.com/vuejs/vue-class-component/issues/337

davidshq committed:
Fix Vue shim, add `artifact_states` to `MIQAConfig` in store

davidshq committed:
Refactoring `ControlPanel` to use Composition API and...

- Adding missing `lockCycle` to `data`
- Comment out non-existent catch handling

davidshq committed:
Making `frame_evaluation` optional on `Frame`, adding `artifact_options` (optional) to `MIQAConfig` and...

Update target version in `tsconfig.json` to `2.7`

davidshq committed:
Upgrading `vue` from `2.6.14` to `2.7.14` which allows us to remove `@vue/composition-api` and `@vue/runtime-dom`

davidshq committed:
Minimal refactoring to remove dependency on `@vue/composition-api` from components and fixing incorrect syntax for `setTaskOverview` in `DecisionButtons.vue`

davidshq committed:
Override TS depend version, upgrade vue router

davidshq committed:
Refactor `hashCode` so conversion of boolean to number is explicit

davidshq committed:
Add additional optional properties to `ResponseData` to eliminate TS complaints

davidshq committed:
Adding types to various variables, ESLint fix

davidshq committed:
Refactor keydown event listener in ControlPanel to use explicit types, use `tagName` instead of `type`

davidshq committed:
Fix issue with app not running when using docker container for npm

davidshq committed:
Add playwright and basic test

davidshq committed:
Minor reformatting fixes for ESLint includes renaming a few components...

- Scan -> ScanView
- Projects -> ProjectsView

Disable Vuetify rules for the moment, changes required too disruptive

davidshq committed:
Add keydown for accessibility

davidshq committed:
Fix decisions on `ScanDecision` to be string not array, specify `last` in `Scan` has a type of `ScanDecision`

Ignoring complaints about multi-word component names for Vue plugins

Minor spelling corrections

davidshq committed:
Adding properties to store that are utilized but not called

Ignore Navbar multi-world lint rule

davidshq committed:
Merge branch 'master' into remove-direct-vuex-2.7

davidshq committed:
Correcting type of duration on WindowLock

davidshq committed:
Move `snackbarAttach` and `promptAttach` into `mounted` to avoid TS complaining.

davidshq committed:
Remove any, replace with explicit import of VRangeSlider

davidshq committed:
Removing playwright tests on pull

davidshq committed:
Removing pyppeteer and codecov from tests

davidshq committed:
Removing with: related to codecov

annehaley committed:
Merge pull request #10 from davidshq/remove-direct-vuex-2.7

Frontend Prep and Updates

zachmullen commented on web_client/.github/workflows/playwright.yml:
```diff
@@ -0,0 +1,25 @@
+    - name: Run Playwright tests
+      run: npx playwright test
+    - uses: actions/upload-artifact@v3
+      if: always()
```
Is this line necessary?

davidshq committed:
Removing unused ColorMaps, updating gitignore, old pyppeteer tox config

davidshq committed:
Updating dependencies

davidshq committed:
Minor code refactorings...

Primarily reformatting for consistency, ESLint compliance
Addition of some explanatory comments
Renaming ExperimentsView's scanState method's state variable to scanTaskState for clarity

davidshq committed:
Refactoring store, Scan...

Renaming Vuex store action `getScan` to `loadScan`, this makes it consistent with other Vuex naming

Renaming method `loadScan` in `Scan.vue` to `swapToScan` to avoid naming conflict and replace call to `getScan` with `loadScan`

davidshq committed:
Removing unused constants, extracting timeout constants into constants.ts, adding TS types to snackbar, renaming `reset` to `continueSession` in `TimeoutDialog`

davidshq committed:
Minor refactoring of ControlPanel

switchLock parameters newExp and oldExp now newExperimentId, oldExperimentId

Additional comments

davidshq committed:
Refactoring of DecisionButtons.vue...

- Removing `currentViewData` from `mapState` b/c (1) it's a getter and (2) it's included in `mapGetters`
- Renaming `dec` in `suggestedArtifacts` to `decision`
- Adding `AUTO_ADVANCE` flag which determines whether scans advance when a decision is made

davidshq committed:
Minor refactoring for VtkViewer...

- Renaming slice parameter `value` to `newSlice`
- Adding type declaration in `takeScreenshot`

davidshq committed:
Refactoring Vuex store...

- Adding a mutation-types.ts file which declares all the mutations we'll be using in the store
- Updating the mutations to standardized naming provided by mutation-types.ts
- Adding type for `proxyManager` param of `shrinkProxyManager` and `prepareProxyManager`
- Adding/updating comments throughout

davidshq committed:
Refactoring Vuex store...

- Removing unused `wholeState` getter
- Refactoring `ADD_SCAN_DECISION` params from `currentScan`, `newDecision` to `currentScanId`, `newScanDecision`
- Refactoring `SET_FRAME_EVALUATION` param from `evaluation` to `frameEvaluation`
- Additional refactorings:
  - SET_LOADING_FRAME: value -> isLoading
  - SET_ERROR_LOADING_FRAME: value -> isErrorLoading
  - ADD_SCAN_FRAME: sid, id -> scanId, frameId
  - ADD_EXPERIMENT_SCANS: eid, sid -> experimentId, scanId
  - ADD_EXPERIMENT: id, value -> experimentId, experiment

davidshq committed:
Refactoring Vuex store...

- Renaming variable `i` in `loadProject` to `experimentIndex`
- Renaming variable `j` in `loadProject` to `scanIndex`
- Renaming variable `k` in `loadProject` to `frameIndex`

davidshq committed:
Refactoring Vuex store...

- Renaming `getArrayName` to `getArrayNameFromFilename`
- Renaming `getData` to `getImageData`
  - Renaming param `id` to `frameId`
- Renaming `loadFile` and `loadFileAndGetData` variable `fileP` to `cachedFile`
- Reduce nesting of async calls in `loadFileAndGetData`
- Renaming `i, j` params in `getNextFrame` to `experimentIndex, scanIndex`

davidshq committed:
Updating calls to Vuex store to use new mutation standardized names, also fixing a problem where crosshair persistence and visibility where not being set correctly

Updating Vuex test to use new mutation names

davidshq committed:
Updating mutation naming in a number of components

davidshq committed:
Updating mutation naming in Projects, fixing sorting issue for `decisionToRating` in ExperimentsView

davidshq committed:
Adding await to `this.refreshTaskOverview()`, `this.login`, `this.loadProject(this.currentProject)`, `store.dispatch('loadProject', newProject)` and `djangoRest.projects`

Updating mutation naming in django.ts

davidshq committed:
Removing unused mutations setCurrentWindowWidth and setCurrentWindowLevel from Vuex store

annehaley committed:
Remove unnecessary file additions

annehaley committed:
Move playwright test to top-level CI

annehaley committed:
Remove outdated vue-utilities directory

annehaley committed:
Manage snackbar with App and store

annehaley committed:
Make all components use composition api and similar structure

annehaley committed:
Type fixes

annehaley committed:
Update packages

annehaley committed:
Behavior fixes

davidshq committed:
Merge pull request #65 from knowledgevis/main-repo-fixes

Switch to only composition API

davidshq committed:
Update github action versions for npm, tox, ci

davidshq committed:
Updating package-lock

davidshq committed:
Test update to CI

davidshq committed:
New ci test

davidshq committed:
Updating package-lock

davidshq committed:
Disabling playwright github action

davidshq committed:
Merge pull request #12 from davidshq/relatively-minor-updates

Refactoring Vuex Store and Other Minor Changes

annehaley commented:
This is now the combination of four different PRs that occurred on the KVis branch

zachmullen commented on .github/workflows/ci.yml:
```diff
@@ -51,45 +51,23 @@ jobs:
+#      with:
+#        name: playwright-report
+#        path: playwright-report/
+#        retention-days: 30
```
Does this not work yet?

annehaley replied:
Correct. The test works fine locally, when the system is already connected to a full database, but we haven't set up such a connection for CI yet, so it fails on an authentication error. Dave suggested we disable it for now.

zachmullen commented on .github/workflows/ci.yml.old:
```diff
@@ -0,0 +1,95 @@
+          DJANGO_MINIO_STORAGE_ENDPOINT: minio:9000
+          DJANGO_MINIO_STORAGE_ACCESS_KEY: minioAccessKey
+          DJANGO_MINIO_STORAGE_SECRET_KEY: minioSecretKey
+          DJANGO_STORAGE_BUCKET_NAME: integration-test-bucket
```
Any reason to keep this around?

annehaley replied:
I suppose this is just there for the history, but if for the sake of cleanup you think we should remove it, we can.

zachmullen replied:
Let's delete any dead code, since we have git to preserve history.

annehaley replied:
Removed in [b76fd50](https://github.com/OpenImaging/miqa/pull/645/commits/b76fd502ccfffb01134feeafe45eb12f9b32111c)

zachmullen commented on web_client/.eslintrc.js:
```diff
@@ -5,12 +5,14 @@ module.exports = {
   },
   extends: [
     'plugin:vue/recommended',
+    // 'plugin:vuetify/recommended',
```
Should this be removed?

annehaley replied:
Sure, I can take that out.

annehaley replied:
Removed in [3acbe32](https://github.com/OpenImaging/miqa/pull/645/commits/3acbe32edb71adb7fbb8d4fe1bc625d183d472fe)

zachmullen commented on web_client/playwright.config.ts:
```diff
@@ -0,0 +1,78 @@
+  //   command: 'npm run start',
+  //   url: 'http://127.0.0.1:3000',
+  //   reuseExistingServer: !process.env.CI,
+  // },
```
Can we remove the commented code?

annehaley replied:
Removed in [3acbe32](https://github.com/OpenImaging/miqa/pull/645/commits/3acbe32edb71adb7fbb8d4fe1bc625d183d472fe)

annehaley committed:
Remove commented lines

annehaley committed:
Remove ci.yml.old file

annehaley committed:
Merge pull request #14 from OpenImaging/kvis-merge-improvements

Pre-merge improvements
